### PR TITLE
feat(api) handle http errors

### PR DESCRIPTION
**Pull request by ZXYmania**
Gives an error message even when the response isn't valid json

### CODE REVIEW

**Review by mat-lord — APPROVED** [review]
Changes look good, pending the above comments.

**Comment by ZXYmania** [comment]
> Can you please run `make format` to format the code correctly

Done

File: tests/test_api.py [hunk 1]
```diff
@@ -194,6 +194,40 @@ def testFailedDelete(self):
...
+        responses.add(
+            responses.GET,
+            f"{Config.api_base_url()}/test/no/error",
+            json=response_dict,
+            status=500,
+        )
+        with self.assertRaises(ApiException):
+            self.Api.get("/test/no/error")
+        
+				# Test an exception which has no json in response
```
[the remark this answers is not in the document]
Comment (ZXYmania) [reply tests/test_api.py:213]: You're talking about using a tab size of 1 right?

[the remark this answers is not in the document]
Comment (ZXYmania) [reply tests/test_api.py]: Made the fix

**Pull request merged**